tests/src/python/test_qgsdatumtransforms.py

```diff
@@ -0,0 +1,148 @@
+
+# -*- coding: utf-8 -*-
+"""QGIS Unit tests for QgsDatumTransforms.
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Nyall Dawson'
+__date__ = '2019-05-25'
+__copyright__ = 'Copyright 2019, The QGIS Project'
+
+from qgis.core import (
+    QgsProjUtils,
+    QgsCoordinateReferenceSystem,
+    QgsDatumTransform
+)
+from qgis.testing import (start_app,
+                          unittest,
+                          )
+from utilities import unitTestDataPath
+
+start_app()
+TEST_DATA_DIR = unitTestDataPath()
+
+
+class TestPyQgsDatumTransform(unittest.TestCase):
+
+    @unittest.skipIf(QgsProjUtils.projVersionMajor() < 6, 'Not a proj6 build')
+    def testOperations(self):
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem(),
+                                           QgsCoordinateReferenceSystem())
+        self.assertEqual(ops, [])
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem('EPSG:3111'),
+                                           QgsCoordinateReferenceSystem())
+        self.assertEqual(ops, [])
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem(),
+                                           QgsCoordinateReferenceSystem('EPSG:3111'))
+        self.assertEqual(ops, [])
+
+        ops = QgsDatumTransform.operations(QgsCoordinateReferenceSystem('EPSG:3111'),
+                                           QgsCoordinateReferenceSystem('EPSG:3111'))
+        self.assertEqual(len(ops), 1)
+        self.assertEqual(ops[0].name, 'Inverse of Vicgrid + Vicgrid')
```

Comment:
I'd suggest not relying too much on the names. This might change in future PROJ versions, particularly in that no-op case